src/manifest/update_periods.ts

```diff
@@ -104,27 +104,38 @@ export function updatePeriods(
                       newPeriods[0],
                       MANIFEST_UPDATE_TYPE.Partial);
 
-  let prevIndexOfNewPeriod = indexOfNewFirstPeriod;
+  let prevIndexOfNewPeriod = indexOfNewFirstPeriod + 1;
   for (let i = 1; i < newPeriods.length; i++) {
     const newPeriod = newPeriods[i];
-    const indexOfNewPeriod = arrayFindIndex(oldPeriods,
-                                            ({ id }) => id === newPeriod.id);
+    let indexOfNewPeriod = -1;
+    for (let j = prevIndexOfNewPeriod; j < oldPeriods.length; j++) {
+      if (newPeriod.id === oldPeriods[j].id) {
+        indexOfNewPeriod = j;
+        break; // end the loop
```

Comment:
`while` instead of using a `break` ?

Reply:
Thought about it. It would be as ugly no?